server/py/framework/db/sqldb/models.py

```diff
@@ -1118,40 +1139,90 @@ def _sqlite_autoincrement(
         target.id = next_id
 
 
-@event_listen_for_dialects(
+@_event_listen_for_dialects(
+    target=AlertActivation,
+    identifier="before_insert",
+    relevant_dialects=[
+        mlrun.common.db.dialects.Dialects.MYSQL,
+        mlrun.common.db.dialects.Dialects.POSTGRESQL,
+    ],
+)
+def set_alert_activations_partition_key(
+    _: Mapper,
+    __: Connection,
+    target: AlertActivation,
+):
+    target.partition_key = target.compute_partition_key()
```

Comment:
I would leave the option to the caller (i.e. the one doing the insert) to specify the partition key manually, so this should check if the `partition_key` is set, and only compute it if empty.
This can have advantages in the future when for example we want to specify that a specific object should not be subject to GC and put it in a special partition that would not be deleted automatically.